src/hooks/useTikTokAuth.ts

```diff
@@ -0,0 +1,155 @@
+import React from 'react'
+
+import moment from 'moment'
+
+// Polyfills
+import 'whatwg-fetch'
+import 'url-search-params-polyfill'
+
+import { IDENTITY_SERVICE } from 'services/AudiusBackend'
+
+type useTikTokAuthOptions = {
+  onError: (e: Error) => void
+}
+
+type WithAuthCallback = (accessToken: string, openId: string) => void
+
+/**
+ * A hook that returns a withAuth function that can be passed a function which will
+ * be provided with the TikTok credentials on existing or successful auth
+ * @param {Object} options
+ * @returns {Function}
+ */
+export const useTikTokAuth = ({
```

Comment:
really like this pattern.

philosophical question - is it really a hook if it doesn't use any of the built in primitive hooks?

Reply:
Interesting question, this could technically just be a function. But because it is effectful in opening a popup I think having it as a hook make sense